build_caldera_api.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Before running: 
+# - go install github.com/go-swagger/go-swagger/cmd/swagger@latest
+
+set -e
+PROJ_PATH="$PWD"
+
+
+CALDERA_VERSION="e4a4b9a320e707e58e69e3213ca07fdc561350ce" # Commit hash
+OUTPUT_BASE_FOLDER="./internal/capability/caldera/api"
+
+OUTPUT="$PROJ_PATH/caldera_api_raw.json"
+
+DOCKER_CMD="podman"
+# DOCKER_CMD="docker"
+
+CALDERA_PORT="8888"
+CALDERA_IMG_NAME="caldera:latest"
+CALDERA_REPO="https://github.com/mitre/caldera"
+TMP_WORKDIR="/tmp/caldera_api_build/"
+
+if [ -d "$TMP_WORKDIR" ]; then
+    rm -rf $TMP_WORKDIR
+fi
+
+mkdir $TMP_WORKDIR
+cd $TMP_WORKDIR
+
+git clone $CALDERA_REPO caldera
+cd caldera/
+git checkout $CALDERA_VERSION
+git submodule init
+git submodule update --recursive
```

Comment:
It feels a bit redundant to actually build the entire image... run it... then remove it again just to obtain the JSON. I get where this comes from, but would like to investigate if we can integrate the process a bit more - the user may want to use Caldera anyway! A good first step probably would be to rely (more) on Docker Compose for this process. I also think it's pretty in line with @MrDurion s comments, to see if we can rely less on scripts outside of Docker.

Reply:
Important to note that this is meant to be able to be executed by a SOARCA *dev* (not a user) once in a blue moon if CALDERA changes their API.

Reply:
Yeah, that's very true, it's mostly about the dev process... it makes this all "probably fine". I'm not too worried about the dependencies. But whereever we can stick it within the existing Docker process we should.